judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/ContestStore.java

```diff
@@ -176,6 +212,94 @@ public Contest createContest(ContestCreateData contestCreateData) {
         });
     }
 
+    public String importDump(ContestDump contestDump) {
+        if (contestDump.getJid().isPresent() && contestDao.selectByJid(contestDump.getJid().get()).isPresent()) {
+            throw ContestErrors.jidAlreadyExists(contestDump.getJid().get());
+        }
+        if (contestDao.selectBySlug(contestDump.getSlug()).isPresent()) {
+            throw ContestErrors.slugAlreadyExists(contestDump.getSlug());
+        }
+
+        ContestModel contestModel = new ContestModel();
+        contestModel.slug = contestDump.getSlug();
+        contestModel.name = contestDump.getName();
+        contestModel.style = contestDump.getStyle().getName().name();
+        contestModel.beginTime = contestDump.getBeginTime();
+        contestModel.duration = contestDump.getDuration().toMillis();
+        contestModel.description = contestDump.getDescription();
+        contestDao.setModelMetadataFromDump(contestModel, contestDump);
+        contestModel = contestDao.persist(contestModel);
+        contestByJidCache.invalidate(contestModel.jid);
```

Comment:
Hmm not sure if these invalidations are required? Since the jid and slug were just created at this point

Reply:
Changed